lib/modules/apostrophe-areas/public/js/editor.js

```diff
@@ -644,25 +645,40 @@ apos.define('apostrophe-areas-editor', {
           'top': '',
           'width': ''
         });
-        $(event.target).after($item);
+        var $oldWidget = $item.find('[data-apos-widget]');
+        var oldEditor = self;
+        var $newArea = $(event.target).closest('[data-apos-area]');
+        var newEditor = $newArea.data('editor');
+        $oldWidget.data('areaEditor', newEditor);
         self.disableDroppables();
-        self.reRenderWidget($item);
-        self.changeOwners($item);
+        oldEditor.startAutosavingThen(function() {
+          $(event.target).after($item);
+          newEditor.startAutosavingThen(function() {
+            newEditor.reRenderWidget($item, function(err) {
+              if (err) {
+                apos.notify('An error occurred.', { type: 'error' });
```

Comment:
Can we also log a more descriptive error on the server?

Reply:
The server-side code hasn't been altered, so I'm going to open an issue for that but move on.